sparkmagic/sparkmagic/controllerwidget/addendpointwidget.py

```diff
@@ -32,21 +33,26 @@ def __init__(self, spark_controller, ipywidget_factory, ipython_display, endpoin
             value='password',
             width=widget_width
         )
+        self.auth_type = self.ipywidget_factory.get_dropdown(
+            options={constants.AUTH_KERBEROS: constants.AUTH_KERBEROS, constants.AUTH_BASIC: constants.AUTH_BASIC,
+                     constants.NO_AUTH: constants.NO_AUTH},
+            description=u"Auth type:"
+        )
```

Comment:
I'm looking at #284, and we had this comment:

>On change, we should hide/show the appropriate fields. So:
>- On change to NO_AUTH: hide username and password.
>- On change to AUTH_KERBEROS: hide username and password.
>- On change to AUTH_SSL: show username and password.

>Ipywidgets has a way to make widgets visible or not:
https://github.com/jupyter-incubator/sparkmagic/blob/master/autovizwidget/autovizwidget/widget/encodingwidget.py#L126